File: Leetcode/leetcode-1207-UniqueNumberOfOccurrences-[array]-[easy]/Solution.java
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        HashMap<Integer, Integer> freq = new HashMap<>();
        for (int num : arr) {
            freq.put(num, freq.getOrDefault(num, 0) + 1);
        }

        HashSet<Integer> frequencies = new HashSet<>();
        for (Integer frequency : freq.values()) {
            if (frequencies.contains(frequency)) {
                return false;
            } else {
                frequencies.add(frequency);
            }
        }

        return true;
    }
}

File: Leetcode/leetcode-1207-UniqueNumberOfOccurrences-[array]-[easy]/Solution.py
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        frequencies = dict({})
        for num in arr:
            frequencies.update({ num: frequencies.get(num, 0) + 1 })
        
        frequencySet = set()
        for freq in frequencies.values():
            if (freq in frequencySet):
                return False
            else:
                frequencySet.add(freq)

        return True